heartbeat/config/config.go

```diff
@@ -41,6 +41,7 @@ type LocationWithID struct {
 // Config defines the structure of heartbeat.yml.
 type Config struct {
 	RunOnce        bool                 `config:"run_once"`
+	PublishTimeout time.Duration        `config:"publish_timeout"`
```

Comment:
nit: would it make sense to move it inside `output.elasticsearch` or its reserved for beats?

Reply:
We enforce it only at heartbeat level, similar to the run_once, so it would be a bit confusing to append it as a global output option.